plone/scale/scale.py

```diff
@@ -187,13 +187,15 @@ def _calculate_all_dimensions(
     if height is not None and (height >= MAX_HEIGHT or height <= 0):
         height = None
 
-    if width is None and height is None:
-        raise ValueError("Either width or height need to be given.")
-
     if mode not in ("contain", "cover", "scale"):
         raise ValueError("Unknown scale mode '%s'" % mode)
 
     dimensions = ScaledDimensions()
+    dimensions.factor_width = dimensions.factor_height = 1.0
+    if width is None and height is None:
+        dimensions.final_width = dimensions.target_width = original_width
+        dimensions.final_height = dimensions.target_height = original_height
+        return dimensions
```

Comment:
It feels weird and brittle to set these attributes right after the `ScaledDimensions` object is initialized.

I would expect `*_width` and `*_height` to be set by the constructor when you call it like `ScaledDimensions(width=original_width, width=original_height)`.
Or maybe we could introduce a new method to do suche a thing (something like `dimension.rescale(width=original_width, width=original_height)`).

Anyway, this is also fine.

Reply:
I have now added an `__init__` method so all properties have an initial value. Feels a bit cleaner indeed. See commit ba783751b4e8514d18e4a865c3f840082a5165df.